dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java

```diff
@@ -42,94 +58,78 @@ public String getName() {
+                Logger.error(this, "There are no Push Publishing environments set to send the bundle.");
             }
-
-            String[] whereToSend = envrions.split(",");
-
-            List<Environment> envsToSendTo = new ArrayList<Environment>();
-            List<Environment> permissionedEnv = new ArrayList<Environment>();
-            List<Environment> finalEnvs = new ArrayList<Environment>();
-
-
+            final String[] whereToSend = environments.split(ENVIRONMENT_DELIMITER);
+            final List<Environment> envsToSendTo = new ArrayList<>();
+            final List<Environment> permissionedEnv = new ArrayList<>();
+            final List<Environment> finalEnvs = new ArrayList<>();
             // Lists of Environments to push to
-            for (String name : whereToSend) {
-                if (UtilMethods.isSet(name)) {
-                    name = name.trim();
-                    final Environment e = APILocator.getEnvironmentAPI().findEnvironmentByName(name);
-                    if (e != null) {
-
-                        envsToSendTo.add(e);
-                    }else{
-                        Logger.error(PushNowActionlet.class, "The Environment " + name + " does not exists");
+            for (String environmentName : whereToSend) {
+                if (UtilMethods.isSet(environmentName)) {
+                    environmentName = environmentName.trim();
+                    final Environment environment = APILocator.getEnvironmentAPI().findEnvironmentByName(environmentName);
+                    if (null != environment && UtilMethods.isSet(environment.getId())) {
+                        envsToSendTo.add(environment);
+                    } else {
+                        Logger.error(this, "The Environment '" + environmentName + "' does not exist.");
                     }
                 }
-
             }
-
-            if(envsToSendTo.isEmpty()){
-                throw new DotPublisherException("There are no enviroments to send the bundle");
+            if (envsToSendTo.isEmpty()) {
+                throw new WorkflowActionFailureException("There are no environments to send the bundle.");
             }
-
-
             // make sure the user has permissions to push
-            boolean isAdmin = APILocator.getUserAPI().isCMSAdmin(processor.getUser());
-            List<Role> roles = APILocator.getRoleAPI().loadRolesForUser(processor.getUser().getUserId(),true);
-            if(isAdmin){
-                List<Environment> app = APILocator.getEnvironmentAPI().findEnvironmentsWithServers();
-                for(Environment e:app)
+            final boolean isAdmin = APILocator.getUserAPI().isCMSAdmin(processor.getUser());
+            final List<Role> roleList = APILocator.getRoleAPI().loadRolesForUser(processor.getUser().getUserId(),true);
+            if (isAdmin) {
+                List<Environment> environmentList = APILocator.getEnvironmentAPI().findEnvironmentsWithServers();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'environmentList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)